tests/components/azure_event_hub/test_client.py

```diff
@@ -0,0 +1,58 @@
+"""Test the client class of AEH."""
+from azure.eventhub.aio import EventHubProducerClient
+import pytest
+
+from homeassistant.components.azure_event_hub.client import (
+    AzureEventHubClient,
+    AzureEventHubClientConnectionString,
+    AzureEventHubClientSAS,
+    ClientCreationError,
+)
+from homeassistant.components.azure_event_hub.const import (
+    CONF_EVENT_HUB_CON_STRING,
+    CONF_EVENT_HUB_NAMESPACE,
+)
+
+from .const import CS_CONFIG_FULL, SAS_CONFIG_FULL
+
+
+@pytest.mark.parametrize(
+    "config, subclass, policy_use, conn_str_use",
+    [
+        (SAS_CONFIG_FULL, AzureEventHubClientSAS, 1, 0),
+        (CS_CONFIG_FULL, AzureEventHubClientConnectionString, 0, 1),
+    ],
+)
+def test_create(
+    config, subclass, policy_use, conn_str_use, mock_policy, mock_from_connection_string
+):
+    """Test the creation of the hubs for sas policy and key."""
+    client = AzureEventHubClient.from_input(**config)
```

Comment:
The client should not be tested directly but indirectly as part of the init and config flow tests.